app/controllers/students_controller.rb

```diff
@@ -2,12 +2,31 @@
 
 class StudentsController < ApplicationController
   before_action :set_student, only: %i[show edit update destroy]
+  before_action :set_year, only: %i[show index]
 
   def index
-    @students = Student.all
+    @students = Course.select('enrollments.code, students.name as student_name, students.id as student_id, courses.name as course_name, courses.year')
+                      .joins("inner join enrollments on courses.id = enrollments.course_id and courses.year = #{@year}")
+                      .joins('right join students on students.id = enrollments.student_id')
   end
 
-  def show; end
+  def show
+    @code, @course_name = Enrollment.select('enrollments.code', 'courses.name')
+                                    .joins(:course)
+                                    .where("courses.year = #{@year} and enrollments.student_id = #{params[:id]}")
```

Comment:
Aqui também é sobre segurança, mas diferente do `joins` aqui é permitido fazer algo assim:
```suggestion
                                    .where("courses.year = ? and enrollments.student_id = ?", @year, params[:id])
```